src/entry.rs

```diff
@@ -504,14 +518,16 @@ impl<'a> EntryFields<'a> {
             || kind.is_gnu_longname()
             || kind.is_gnu_longlink()
         {
-            return Ok(());
+            return Ok(Unpacked::__Nonexhaustive);
         };
 
         // Old BSD-tar compatibility.
         // Names that have a trailing slash should be treated as a directory.
         // Only applies to old headers.
         if self.header.as_ustar().is_none() && self.path_bytes().ends_with(b"/") {
-            return self.unpack_dir(dst);
+            self.unpack_dir(dst)?;
+            // Why no permission setting?
```

Comment:
This is probably just a bug and you can feel free to update this

Reply:
done